documentation/asciidoc/computers/raspberry-pi/raspberry-pi-industrial.adoc

```diff
@@ -92,7 +92,63 @@ It is possible to prevent the customer OTP bits from being read at all. This can
 
 [,bash]
 ----
-/opt/vc/bin/vcmailbox 0x00038021 8 8 0xffffffff 0xaffebabe
+vcmailbox 0x00038021 8 8 0xffffffff 0xaffebabe
 ----
 
 This operation is unlikely to be useful for the vast majority of users, and is irreversible.
+
+=== Device specific private key
+Eight rows of OTP (256bits) are available for use as a device-specific private key. This is intended to support file-system encryption.
+
+These rows can be programmed and read using similar `vcmailbox` commands to those used for managing customer OTP rows. If
+secure-boot / file-system encryption is not required then the device private key rows can be used to store general purpose information.
+
+* The device private key rows can only be read via the `vcmailbox` command which requires access to `/dev/vcio` which is restricted to the `video` group on Raspberry Pi OS.
+* Raspberry Pi computers do not have a hardware protected key store.  It is recommended that this feature is used in conjunction with https://github.com/raspberrypi/usbboot/blob/master/secure-boot-example/README.md[secure-boot] in order to restrict access to this data.
+* Raspberry Pi OS does not support an encrypted root-filesystem.
+
+See https://gitlab.com/cryptsetup/cryptsetup[cryptsetup] for more information about open-source disk encryption.
+
+==== Key programming script `rpi-otp-private-key`
+The https://github.com/raspberrypi/usbboot/blob/master/tools/rpi-otp-private-key[rpi-otp-private-key] script wraps the device private key `vcmailbox` APIs in order to make it easier to read/write a key in the same format as OpenSSL.
+
+Read the key as a 64-byte hex number
+[,bash]
+----
+rpi-otp-private-key
+----
+
+Example output
+----
+f8dbc7b0a4fcfb1d706e298ac9d0485c2226ce8df7f7596ac77337bd09fbe160
+----
+
+Writes a 64-byte randomly generated number to the device private key. +
+**Warning: This operation cannot be undone. **
+[,bash]
+----
+# rpi-otp-private-key -w $(openssl rand -hex 32)
```

Comment:
If the wrapper script does everything needed, does it really make sense to document the underlying low-level commands? If anybody actually needs that "raw" vcmailbox functionality, they could just look at the source of the script?
(which seems especially pertinent as any OTP write-actions are irreversible)

Reply:
Yes. I expect people will roll their own and we should document thr mailbox APIs